master/internal/api_runs.go

```diff
@@ -306,6 +308,10 @@ func (a *apiServer) MoveRuns(
 		return nil, status.Error(codes.PermissionDenied, err.Error())
 	}
 
+	if req.SourceProjectId == req.DestinationProjectId {
```

Comment:
Good catch! This will avoid unnecessary increments of the local run ids & bogging up the redirects table!